### ISSUE TITLE
Breakpoints are not working when connected to maya PyCharm 2020.3

### ISSUE THREAD

**Issue by chadlichty** [cmcpasserby/MayaCharm#56]
Here is the output from pycharm after connecting to maya 2020:
I connecting by attaching to the maya process.
```
D:\VasgApps\maya\2020\maya2020_pfix_16_0\bin\mayapy.exe C:\Users\clichty\AppData\Local\JetBrains\PyCharm2020.3\MayaCharm\vasg_infrastructure.4a9b6bfa\python\attach_pydevd.py --port 59764 --pid 3320 --mcPort 4437 --pydevPath "C:\Program Files\JetBrains\PyCharm 2020.3.3\plugins\python\helpers\pydev"
Connected to pydev debugger (build 203.7148.72)
```
If I call print("Hello World") in maya's script editor I get the print out in Maya and pycharm.

The script that has the breakpoints is then launched from within maya.  I normally use a tool shelf button in maya but it can be launched from within the script editor.  The script lives locally on one of my hard drives.  Is that what you meant by where the script is located.  

I tested this on another machine and the connection is being refused. So I was not able to test 
```
D:\VasgApps\Maya\2020\maya2020_pfix_16_0\bin\mayapy.exe C:\Users\clichty\AppData\Local\JetBrains\PyCharm2020.3\MayaCharm\infrastructure.f88d2eec\python\attach_pydevd.py --port 59241 --pid 19972 --mcPort 4434 --pydevPath "C:\Program Files\JetBrains\PyCharm 2020.3.3\plugins\python\helpers\pydev"
Traceback (most recent call last):
  File "C:\Users\clichty\AppData\Local\JetBrains\PyCharm2020.3\MayaCharm\infrastructure.f88d2eec\python\attach_pydevd.py", line 88, in <module>
    main(process_command_line(sys.argv[1:]))
  File "C:\Users\clichty\AppData\Local\JetBrains\PyCharm2020.3\MayaCharm\infrastructure.f88d2eec\python\attach_pydevd.py", line 84, in main
    send_command(setup['mcPort'], python_code)
  File "C:\Users\clichty\AppData\Local\JetBrains\PyCharm2020.3\MayaCharm\infrastructure.f88d2eec\python\attach_pydevd.py", line 54, in send_command
    client.connect(("localhost", port))
```
  File "D:\VasgApps\Maya\2020\maya2020_pfix_16_0\bin\python27.zip\socket.py", line 228, in meth
socket.error: [Errno 10061] No connection could be made because the target machine actively refused it

Process finished with exit code 0
Server stopped.

**Comment by cmcpasserby** [comment]
if you just the run config to send maya a command or script to execute does that work?

**Comment by chadlichty** [comment]
Yes If I run print(hello) it does print hello in the maya script editor
![image](https://user-images.githubusercontent.com/14267500/111378749-7f425200-865f-11eb-8dcd-b87c98dca6ea.png)

**Comment by cmcpasserby** [comment]
looks like the command port is not attached or setup, did you add or run the proper stuff in maya to setup the command port, and are things setup for your maya instance in Settings?

**Comment by chadlichty** [comment]
I have updated the userSetup.py with
```
import maya.cmds as cmds

if not cmds.commandPort(":4437", query=True):
    cmds.commandPort(name=":4437")
```
I also have tried running that in the script editor.  Is there anything else that needs to be done?
I do have added multiple maya sdks to the mayaCharm settings.
![image](https://user-images.githubusercontent.com/14267500/111381345-d0a01080-8662-11eb-998a-fe684c1328f6.png)

**Comment by cmcpasserby** [comment]
so after doing this, does trying to execute code via a run config work?

**Comment by chadlichty** [comment]
No, it does not work. I had already tried all of that before posting that I had an issue. I tried again and nothing has changed.

```
D:\VasgApps\Maya\2020\maya2020_pfix_16_0\bin\mayapy.exe C:\Users\clichty\AppData\Local\JetBrains\PyCharm2020.3\MayaCharm\infrastructure.f88d2eec\python\attach_pydevd.py --port 63440 --pid 19972 --mcPort 4434 --pydevPath "C:\Program Files\JetBrains\PyCharm 2020.3.3\plugins\python\helpers\pydev"
Traceback (most recent call last):
  File "C:\Users\clichty\AppData\Local\JetBrains\PyCharm2020.3\MayaCharm\infrastructure.f88d2eec\python\attach_pydevd.py", line 88, in <module>
    main(process_command_line(sys.argv[1:]))
  File "C:\Users\clichty\AppData\Local\JetBrains\PyCharm2020.3\MayaCharm\infrastructure.f88d2eec\python\attach_pydevd.py", line 84, in main
    send_command(setup['mcPort'], python_code)
  File "C:\Users\clichty\AppData\Local\JetBrains\PyCharm2020.3\MayaCharm\infrastructure.f88d2eec\python\attach_pydevd.py", line 54, in send_command
    client.connect(("localhost", port))
  File "D:\VasgApps\Maya\2020\maya2020_pfix_16_0\bin\python27.zip\socket.py", line 228, in meth
socket.error: [Errno 10061] No connection could be made because the target machine actively refused it
```

**Comment by cmcpasserby** [comment]
k, either a firewall or something is in the way, or you found a bug and its trying to attach to the wrong instance of Maya.
in the command args you can see its using `--mcPort 4434`, but you are trying to connect to your maya 2020 at 4437 correct?

**Comment by chadlichty** [comment]
Same thing with Maya2018.  When the userSetup.py script is not run the list of available programs to attach to doesn't have the maya section.  So that part seems to be working correctly because the maya section is there at the top making it easy to select maya to attach to.

```
D:\VasgApps\Maya\2018\maya2018_x64_sie-4.2\bin\mayapy.exe C:\Users\clichty\AppData\Local\JetBrains\PyCharm2020.3\MayaCharm\infrastructure.f88d2eec\python\attach_pydevd.py --port 63651 --pid 13080 --mcPort 4434 --pydevPath "C:\Program Files\JetBrains\PyCharm 2020.3.3\plugins\python\helpers\pydev"
Traceback (most recent call last):
  File "C:\Users\clichty\AppData\Local\JetBrains\PyCharm2020.3\MayaCharm\infrastructure.f88d2eec\python\attach_pydevd.py", line 88, in <module>
    main(process_command_line(sys.argv[1:]))
  File "C:\Users\clichty\AppData\Local\JetBrains\PyCharm2020.3\MayaCharm\infrastructure.f88d2eec\python\attach_pydevd.py", line 84, in main
    send_command(setup['mcPort'], python_code)
  File "C:\Users\clichty\AppData\Local\JetBrains\PyCharm2020.3\MayaCharm\infrastructure.f88d2eec\python\attach_pydevd.py", line 54, in send_command
    client.connect(("localhost", port))
  File "D:\VasgApps\Maya\2018\maya2018_x64_sie-4.2\bin\python27.zip\socket.py", line 228, in meth
socket.error: [Errno 10061] No connection could be made because the target machine actively refused it
```

**Comment by chadlichty** [comment]
ok so Let me try port 4434 looks like all of the mayacharm settings are using the same port number.

**Comment by chadlichty** [comment]
yep changing the port worked for my one machine.  It connects and breakpoints are caught by the debugger. My other machine was connecting fine but not catching the break points.

**Comment by chadlichty** [comment]
Thanks!

**Comment by cmcpasserby** [comment]
k, yeah its a bug then, the port number needed when creating the command port should match up with what you see in its settings. I will try to find a way to reproduce even know i only have access to 1 Maya version and failing that i will give you a build with some more logging enabled so i can use those logs to see what is going on.

**Comment by chadlichty** [comment]
sounds good. 
you should sign up to be an Autodesk developer. Let them know you the creator of mayacharm and you need access to maya version for testing.  It took a little bit to get done but now I have access to all autodesk products.

**Comment by chadlichty** [comment]
of course i'm available to do testing.

**Comment by cmcpasserby** [comment]
while you have the wanted instance of maya running, can you run this command in the terminal
`wmic process where 'ProcessID=7100' get ExecutablePath`
replace the 7100 with the PID for your Maya instance

**Comment by cmcpasserby** [comment]
actrully don't worry about getting the output of that command, found the issue. no matter what you were connecting too, it was only using the port number for the version selected as the Active Maya Sdk in the settings. will get a fix out in a day or so that fixes this.

**Comment by chadlichty** [comment]
AWESOME.  You seriously need to set a patreon account or something so techartists can support your efforts with some loot.

**Comment by cmcpasserby** [comment]
just give me a heads up if this test build fixes the issue
[MayaCharm-3.2.2-test.1.zip](https://github.com/cmcpasserby/MayaCharm/files/6153290/MayaCharm-3.2.2-test.1.zip)

**Comment by chadlichty** [comment]
Works great, thanks!

**Comment by cmcpasserby** [comment]
cool, tonight i will do a few more tests and publish it. generally takes 2 weekdays from being published to get approval and go live.

**Comment by cmcpasserby** [comment]
Released [3.2.2](https://github.com/cmcpasserby/MayaCharm/releases/tag/v3.2.2) should be in the plugin marketplace in a day or 2.
